src/Pages/7 Writing Plugins/Testing.md

```diff
@@ -0,0 +1,312 @@
+---
+title: 'Testing Your Plugin'
+date: 2025-01-15
+image: images/blake-logo.png
+tags: []
+description: "Comprehensive testing strategies for Blake plugins including unit tests, integration tests, and development tips."
+iconIdentifier: "bi bi-plus-square-fill-nav-menu"
+pageOrder: 8
+category: "Writing Plugins"
+---
+
+:::summary
+Test Blake plugins with unit tests for plugin logic, integration tests with actual Blake content, and thorough development testing across various content types and configurations.
+:::
+
+## Testing Your Plugin
+
+### Unit Testing
+
+Test your plugin logic independently of Blake:
+
+```csharp
+[Fact]
+public async Task LastModifiedPlugin_AddsTimestampMetadata()
+{
+    // Arrange
+    var plugin = new LastModifiedPlugin();
+    var context = CreateTestBlakeContext();
+    var logger = new TestLogger();
+
+    // Act
+    await plugin.BeforeBakeAsync(context, logger);
+
+    // Assert
+    Assert.All(context.Pages, page =>
+        Assert.True(page.Metadata.ContainsKey("lastModified")));
+}
+```
+
+**Creating Test Contexts:**
```

Comment:
Please review the code in the source Blake repo. This is not correct for how to construct a BlakeContext instance.

Note to self: should I create a public BlakeContextGenerator or test helper? Can use it in the Blake tests, and all plugin tests. Template tests too.